src/Fhir.Anonymizer.Core.UnitTests/Processors/CryptoHashProcessorTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Fhir.Anonymizer.Core.Processors;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Xunit;
+
+namespace Fhir.Anonymizer.Core.UnitTests.Processors
+{
+    public class CryptoHashProcessorTests
+    {
+        private const string TestHashKey = "123";
+
+        public static IEnumerable<object[]> GetNonReferenceNodesForCryptoHash()
+        {
+            yield return new object[] { new Id(string.Empty), string.Empty };
```

Comment:
Curious, can there be a resource with empty id (as per spec)?

Reply:
Resource ids can not be empty. Yet users can config other field to cryptoHash method like valueString, description, display etc. And if given an empty case, we will handle the empty case to an empty result.